db/commitlog/commitlog.cc

```diff
@@ -1533,14 +1542,21 @@ future<> db::commitlog::segment_manager::clear_reserve_segments() {
         _reserve_segments.pop();
     }
 
-    auto re = std::exchange(_recycled_segments, {});
-    auto i = re.begin();
-    auto e = re.end();
+    std::vector<sstring> tmp;
+    tmp.reserve(_recycled_segments.size());
+
+    _recycled_segments.consume([&](sstring s) {
+        tmp.emplace_back(std::move(s));
+        return true;
+    });
+
+    auto i = tmp.begin();
+    auto e = tmp.end();
 
     return parallel_for_each(i, e, [this](const sstring& filename) {
         clogger.debug("Deleting recycled segment file {}", filename);
         return delete_file(filename);
-    }).finally([this, re = std::move(re)] {
+    }).finally([this, tmp = std::move(tmp)] {
         return do_pending_deletes();
     });
```

Comment:
Not sure why we are deleting those recycled segments. Won't we need to recreate them on the next startup?

Note, this isn't a comment on the patch but on the surrounding code.

Reply:
This is a sort of round-about mechanism, but the do_pending_deletes will go to delete_segments which will in fact probably just add most of these files to a new recycled list. The only point with this exercise is that we go through disk footprint checks one final time, only keeping the files that are under the limit. So in the new world where we uphold this limit, the above re-recycling should be pretty pointless.